src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -130,10 +131,14 @@ static Schema preProcessSchema(Schema schema) {
         Schema valueSchema = schema.valueSchema();
         String keyName = keySchema.name() == null ? keySchema.type().name() : keySchema.name();
         String valueName = valueSchema.name() == null ? valueSchema.type().name() : valueSchema.name();
-        Schema elementSchema = SchemaBuilder.struct().name(keyName + "-" + valueName)
-            .field(MAP_KEY, preProcessSchema(keySchema))
-            .field(MAP_VALUE, preProcessSchema(valueSchema))
-            .build();
+        Schema elementSchema;
```

Comment:
No need for early declaration. `elementSchema` is not used in the block for String types.